singularity-eos/eos/get_sg_eos_functors.hpp

```diff
@@ -216,12 +277,144 @@ struct final_functor {
     if (do_s) {
       sie_v(i) /= mass_sum;
     }
+    check_all_vals(i, npte, tid, pte_converged);
     /* reset mass fractions to original values if not normalized to 1 */
     for (int m = 0; m < nmat; ++m) {
       frac_mass_v(i, m) *= mass_sum;
     }
     return;
   }
+
+ private:
+  // Check to make sure returned values are normal or zero
+  template <typename valT>
+  PORTABLE_FORCEINLINE_FUNCTION int bad_val(valT const value,
```

Comment:
What is the purpose of duplicating this function across objects? If it's common then it should be moved up scope and have a single definition.

Reply:
The print message is different between the two definitions in the two functors. I can probably generalize it better though and lift the functionality up in scope

Reply:
I don't want to hold things up, I just get flustered seeing duplicated code. It can be changed later.

Reply:
I would also prefer this be more generic. IMO something like this should live in a utility header.

Reply:
Added a utility header for error utilities. I also followed @mauneyc-LANL 's suggestion and broke out the condition from the message.

I had originally wanted the value checks to print out different things if they were in the input versus output, but I realize that I'm covering that in my error message so the checks themselves can be generic.